src/main/java/com/conveyal/datatools/common/utils/Scheduler.java

```diff
@@ -1,63 +1,203 @@
 package com.conveyal.datatools.common.utils;
 
-import com.conveyal.datatools.manager.DataManager;
-import com.conveyal.datatools.manager.controllers.api.ProjectController;
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.jobs.FeedExpirationNotificationJob;
+import com.conveyal.datatools.manager.jobs.FetchProjectFeedsJob;
 import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.FeedVersion;
 import com.conveyal.datatools.manager.models.Project;
 import com.conveyal.datatools.manager.persistence.Persistence;
 import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.ListMultimap;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
 import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static com.conveyal.datatools.common.utils.Utils.getTimezone;
 
 public class Scheduler {
-    private static final Logger LOG = LoggerFactory.getLogger(DataManager.class);
+    private static final Logger LOG = LoggerFactory.getLogger(Scheduler.class);
 
-    // Stores ScheduledFuture objects that kick off runnable tasks (e.g., fetch project feeds at 2:00 AM).
-    public static Map<String, ScheduledFuture> autoFetchMap = new HashMap<>();
     // Scheduled executor that handles running scheduled jobs.
     public final static ScheduledExecutorService schedulerService = Executors.newScheduledThreadPool(1);
-    // multimap for keeping track of notification expirations
-    public final static ListMultimap<String, Future> scheduledNotificationExpirations = ArrayListMultimap.create();
+    /** Maps {@link ScheduledJob} objects containing scheduled tasks to the ID associated with the originating object. */
+    public final static ListMultimap<String, ScheduledJob> scheduledJobsById = ArrayListMultimap.create();
```

Comment:
I'm not a fan of using a plain old string as the key here.  The likelihood of a collision of ids is extremely small, but it just seems to cause a potential for mismatching things when some ids are project ids and others are feed source ids.  I'd be more comfortable if there was a way to have a key that also said what type of object the id was associated with.

Reply:
I don't think we need to worry about id collision and I'm not sure how a mismatch would occur. I do agree that this doesn't feel exactly right, but I'm not sure changing the key to include the object type would improve this code. Maybe @abyrd has an idea.

Reply:
I really do think including a reference to the object type would improve it though and future-proof the code.  For example, what if two entity types use the same ScheduledJob type and we only want to delete the notifications associated with a specific entity type?  With the current code they'd all potentially get deleted.  Furthermore if we wanted to analyze the characteristics of all scheduled jobs we may not have a way of determining what the id is associated with.  This feels very wrong, but I don't have a textbook answer for why.